rstcheck/__init__.py

```diff
@@ -516,18 +525,48 @@ def load_configuration_from_file(directory: str, args: argparse.Namespace) -> ar
 
 
 def _get_options(directory_or_file: str, debug: bool = False) -> typing.Dict[str, str]:
+    options = {}
+
     config_path = find_config(directory_or_file, debug=debug)
-    if not config_path:
```

Comment:
Early return are a good practice especially if you have an error or nothing to do. If we have no config file we have nothing to do and can return early. Why run the other if statements too? This only increases runtime. Therefore the original early return was very good.

Here is what I would do with this function
```python
def _get_options(directory_or_file: str, debug: bool = False) -> typing.Dict[str, str]:
    config_path = find_config(directory_or_file, debug=debug)
    if not config_path:
        return {}

    if pathlib.Path(config_path).name == "pyproject.toml":
        return _get_pyproject_options(config_path)

    parser = configparser.ConfigParser()
    parser.read(config_path)
    try:
        return dict(parser.items("rstcheck"))
    except configparser.NoSectionError:
        return {}
```
Here we have early returns, which makes the code more readable IMO and shorter too. The nesting also decreases, which also improves readability.

If you use pathlib.Path or os.path does not make a big difference. I personally think pathlib.Path is more readable and I like it's API and handling more. Personal preference.